crates/syntax/src/ast/make.rs

```diff
@@ -149,8 +149,13 @@ pub fn impl_(
     ast_from_text(&format!("impl{} {}{} {{}}", params, ty, ty_params))
 }
 
-pub fn impl_trait(trait_: ast::Path, ty: ast::Path) -> ast::Impl {
-    ast_from_text(&format!("impl {} for {} {{}}", trait_, ty))
+pub fn impl_trait(
+    trait_: ast::Path,
+    ty: ast::Path,
+    ty_params: Option<ast::GenericParamList>,
+) -> ast::Impl {
+    let ty_params = ty_params.map_or_else(String::new, |params| params.to_string());
+    ast_from_text(&format!("impl{2} {} for {}{2} {{}}", trait_, ty, ty_params))
```

Comment:
Slightly incorrect, `ast::GenericParamList` also contains bounds which are invalid in the second use of the `ty_params` here in the formatting string. See `generate_impl_text` in `crates/ide_assists/src/utils.rs` which also does something similar(might be possible to reuse that function here as well somehow, as its already used in other branches in this assist)

Reply:
Should I just copy the code from `generate_trait_impl_text` because there is no way to use that function here?

Reply:
If you can't make use of it as is, then ye just copy the things you need.